wled00/bus_manager.h

```diff
@@ -73,6 +73,62 @@ struct BusConfig {
   }
 };
 
+// Defines an LED Strip and its color ordering.
+struct ColorOrderMapEntry {
+  uint16_t start;
+  uint16_t len;
+  uint8_t colorOrder;
+};
+
+struct ColorOrderMap {
+  void add(uint16_t start, uint16_t len, uint8_t colorOrder) {
+    if (_count >= WLED_MAX_COLOR_ORDER_MAPPINGS) {
+      return;
+    }
+    if (len == 0) {
+      return;
+    }
+    if (colorOrder > COL_ORDER_MAX) {
+      return;
+    }
+    _mappings[_count].start = start;
+    _mappings[_count].len = len;
+    _mappings[_count].colorOrder = colorOrder;
+    _count++;
+  }
+
+  uint8_t count() const {
+    return _count;
+  }
+
+  void reset() {
+    _count = 0;
+    memset(_mappings, 0, sizeof(_mappings));
+  }
+
+  const ColorOrderMapEntry* get(uint8_t n) const {
+    if (n > _count) {
+      return nullptr;
+    }
+    return &(_mappings[n]);
+  }
+
+  inline uint8_t getPixelColorOrder(uint16_t pix, uint8_t defaultColorOrder) const {
```

Comment:
I am fond of inline functions but I don't see the benefit here.

Reply:
My thinking was that, given that `getPixelColorOrder()`  is called each time a pixel's color is set, inlining `getPixelColorOrder()` would improve performance by eliminating the overhead of having to call and return from the function.